parser-typechecker/src/Unison/Codebase/Causal/Type.hs

```diff
@@ -0,0 +1,139 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Unison.Codebase.Causal.Type
+  ( Causal (..),
+    RawHash (..),
+    pattern One,
+    pattern Cons,
+    pattern Merge,
+    before,
+    children,
+    lca,
+    head_,
+  )
+where
+
+import qualified Control.Lens as Lens
+import qualified Data.Map as Map
+import qualified Data.Sequence as Seq
+import qualified Data.Set as Set
+import Unison.Hash (Hash)
+import Unison.Prelude
+import Prelude hiding (head, read, tail)
+
+{-
+`Causal a` has 5 operations, specified algebraically here:
```

Comment:
Love the descriptions, but maybe these should just be haddocks (at the very least a module doc, so it shows up in haddocks?)
I'm worried about this description, or the contained types, going out of date.
Case-in-point, the `Causal` type in the descriptions switches between `Causal a` and `Causal m a`, when in fact the type is `Causal m h a` 🤔